File: backend/app/schemas/task.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel

from app.models.task import TaskStatus


class TaskCreate(BaseModel):
    name: str
    task_type: str
    model_id: uuid.UUID | None = None
    dataset_id: uuid.UUID | None = None
    config: dict[str, Any] | None = None


class TaskResponse(BaseModel):
    id: uuid.UUID
    name: str
    task_type: str
    status: TaskStatus
    model_id: uuid.UUID | None
    dataset_id: uuid.UUID | None
    config: dict[str, Any] | None
    progress: int
    result: dict[str, Any] | None
    error_message: str | None
    started_at: datetime | None
    finished_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TaskArtifactItem(BaseModel):
    key: str
    filename: str
    url: str


class TaskArtifactsResponse(BaseModel):
    items: list[TaskArtifactItem]

File: frontend/src/types/task.ts
export interface Task {
  id: string
  name: string
  task_type: string
  status: string
  model_id: string | null
  dataset_id: string | null
  dataset_version_id: string | null
  dataset_export_id: string | null
  config: Record<string, unknown> | null
  progress: number
  result: Record<string, unknown> | null
  error_message: string | null
  started_at: string | null
  finished_at: string | null
  created_at: string
  updated_at: string
}

export interface TaskArtifactItem {
  key: string
  filename: string
  url: string
}

export interface TaskHistoryPoint {
  epoch: number
  train_loss?: number
  map50?: number
  map50_95?: number
}

export interface TrainingDetailContext {
  datasetName: string
  modelName: string
  modelVersion: string
  versionName: string
  exportName: string
  exportFormat: string
  dataYamlPath: string
  exportCreatedAt: string
  exportNotes: string
  splitSummary: string
  classCount: number
  classNames: string[]
  imageCount: number
  boxCount: number
  splitCounts: Record<string, number>
}
